<a href="https://colab.research.google.com/github/SDlel/ml-assignments/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/SDlel/ml-assignments/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

I chose the Refresh / Content Opportunity Scoring lane because it focuses on a practical problem that content teams face, deciding which pages should be reviewed first. Instead of relying only on fixed rules, I want to explore whether search and engagement data can help create a better ranking of pages that deserve attention. This lane also builds naturally on the starter notebooks, so it gives me a solid foundation for the rest of the internship.

In [10]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
!git clone https://github.com/SDlel/ml-assignments.git
%cd ml-assignments

Cloning into 'ml-assignments'...
remote: Enumerating objects: 119, done.
remote: Counting objects: 100% (119/119), done.
remote: Compressing objects: 100% (74/74), done.
remote: Total 119 (delta 36), reused 95 (delta 29), pack-reused 0 (from 0)
Receiving objects: 100% (119/119), 1.84 MiB | 13.78 MiB/s, done.
Resolving deltas: 100% (36/36), done.
/content/ml-assignments/ml-assignments


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

The decision I want to improve is which content pages should be reviewed first for a refresh. The people taking action would be content managers or SEO teams who decide whether a page should be updated, expanded, monitored, or left as it is. A wrong recommendation could waste time reviewing pages that do not need attention, or it could miss pages that are actually declining and should have been reviewed sooner. I hope that using search and engagement data can make these decisions more consistent and evidence based.

In [11]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Rows:", len(df))
print("Unique clients:", df["client_id"].nunique())
print("Average 90-day impressions:", round(df["impressions_90d"].mean(), 2))
print("Pages trending down:", (df["trend_direction"] == "down").sum())

Rows: 30000
Unique clients: 32
Average 90-day impressions: 5200.37
Pages trending down: 16262


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

I loaded the starter dataset to understand whether this lane is worth exploring further. The dataset contains 30,000 content pages from 32 unique clients, which provides enough variety to study content performance across different websites. The average page received about 5,200 impressions over the last 90 days, showing that many pages have meaningful search visibility. I also found that 16,262 pages are labeled as trending down, suggesting there are many potential candidates for refresh and making this a useful problem to investigate over the next several weeks.

In [12]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project will use observed search and engagement data to help prioritize which pages should be reviewed first. Any results I produce should be treated as decision support rather than proof that a page will improve after being refreshed. I cannot claim that my model proves why a page gained or lost traffic, predicts Google's ranking algorithm, or shows that refreshing a page directly caused better performance. Any recommendations will still require human review before action is taken.

In [13]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.